fdbserver/storageserver.actor.cpp

```diff
@@ -2068,6 +2068,7 @@ ACTOR Future<Void> fetchKeys( StorageServer *data, AddingShard* shard ) {
 						shard->server->addShard( ShardInfo::addingSplitLeft( KeyRangeRef(keys.begin, nfk), shard ) );
 						shard->server->addShard( ShardInfo::newAdding( data, KeyRangeRef(nfk, keys.end) ) );
 						shard = data->shards.rangeContaining( keys.begin ).value()->adding;
+						warningLogger = logFetchKeysWarning(shard);
```

Comment:
I agree this solves the segmentation fault.
To ensure the `logFetchKeysWarning` is ever called in simulation, we need to make the `wait(delay(600));`  in `logFetchKeysWarning`  in simulation.